### PR TITLE
Instances of BaseException to be replaced with Exception.

### PR DESCRIPTION
**Pull request by prmukherj**
Resolved several instances of except BaseException

### CODE REVIEW

**Review by hpohekar — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

File: tests/test_casereader.py [hunk 1]
```diff
@@ -202,7 +202,7 @@ def test_case_reader_get_rp_and_config_vars():
     assert reader.rp_var.pressure.output_dpdt__q() is True
     assert len(reader.rp_var.context.map_r17__plus()) == 53
     assert reader.rp_var.defaults.pre_r19__dot0_early__q() is False
-    with pytest.raises(BaseException):
+    with pytest.raises(Exception):
```
Comment (mkundu1) [inline tests/test_casereader.py:205]: generally better to test the specific exception which is thrown in this case

**Review by mkundu1 — APPROVED** [review]

File: tests/test_casereader.py (hunk 1 above)
Comment (raph-luc) [reply tests/test_casereader.py:205]: Not only in tests, but ideally all of the exception catches should be as specific as possible

Regardless, not criticizing as this is still a big improvement, thanks @prmukherj

**Review by raph-luc — APPROVED** [review]

File: src/ansys/fluent/core/launcher/launcher.py [hunk 2]
```diff
@@ -393,8 +393,8 @@ def _get_running_session_mode(
                 if fluent_connection.scheme_eval.scheme_eval("(cx-solver-mode?)")
                 else "meshing"
             )
-        except BaseException:
-            raise RuntimeError("Fluent session password mismatch")
+        except Exception as ex:
+            raise RuntimeError("Fluent session password mismatch") from ex
```
Comment (raph-luc) [inline src/ansys/fluent/core/launcher/launcher.py:397]: Good catch

File: tests/test_casereader.py (hunk 1 above)
Comment (prmukherj) [reply tests/test_casereader.py]: Done, Thank you @mkundu1.

Thank you @raph-luc, I totally agree with the point you raised. However, covering all the specific exceptions for the cases where these changes have been made seems a bit tricky. That's why for now I think we can re-raise the Exception using the from keyword which provides user the information about the original error as well. Seems fine for now, right?

Comment (raph-luc) [reply tests/test_casereader.py]: That is right, I fully agree

**Pull request merged**